src/test.rs

```diff
@@ -150,6 +150,12 @@ mod marlin {
             // Fail verification
             assert!(!Marlin::<F, PC, D>::verify(&index_vk, &pc_vk, &[a, a], &proof).unwrap());
 
+            // Fail verification using randomized proof
+            let mut fake_proof = proof.clone();
+            fake_proof.randomize_commitments();
+            println!("\nShould not verify (i.e. verifier messages should print below):");
+            assert!(!Marlin::<F, PC, D>::verify(&index_vk, &pc_vk, &[c, d], &fake_proof).unwrap());
```

Comment:
Mmm it seems that this still fails in the AHP part ("Outer sumcheck"). I would like it to fail at the level of opening proof. Can you change it ? Also can you explicitly change the assertion to assert error and assert that the error is in the opening proof verification ?

Reply:
I've changed the code so it now fails at the opening, but I cannot put correct assertion in test. The "verify" function wrap all the errors and returns only "Ok(false)"